coalib/misc/Caching.py

```diff
@@ -0,0 +1,171 @@
+import os
+import calendar
+import time
+import appdirs
+import pickle
+
+
+from coalib.misc.Constants import caching_db, changed_files_db
+
+
+def get_config_data_path(filename):
+    """
+    Gets the full path of ``filename`` present in the user config directory.
+
+    :param filename: The file whose path needs to be expanded.
+    :return:         Full path of the file, assuming it's present in the
+                     user's config directory.
+    """
+    path = appdirs.user_config_dir("coala")
+    return os.path.join(path, filename)
+
+
+def pickle_load(filename, fallback=None):
+    """
+    Gets the data stored in ``filename`` present in the coala user
+    config directory.
+
+    :param filename: The name of the file present in the coala user config
+                     directory.
+    :param fallback: Return value to fallback to in case the file doesn't
+                     exist.
+    :return:         Data that is present if the file, if the file exists.
```

Comment:
@hypothesist Seems typo here.
I think `Data that is present in the file, if the file exists` is what you wanted? Upto you.